### ISSUE TITLE
Tracking issue to support Redis and SQLite3 

### ISSUE THREAD

**Issue by sdww0** [asterinas/asterinas#1048]
### Description

This is the tracking issue to support Redis and SQLite3. I run Redis (Version 7.2.0-v11) and SQLite3 (with [the benchmark in LevelDB](https://github.com/google/leveldb/blob/main/benchmarks/db_bench_sqlite3.cc)) successfully in my [`dev-apps`](https://github.com/sdww0/asterinas/tree/dev-apps) branch. There remain serval gaps to run Redis and SQLite3.

### Current Status

The fixes I have made:

- User heap limit is too small. I extend it to 64MB temporarily.
- Ignore timeout error in `epoll_wait` syscall. As the [manual](https://www.man7.org/linux/man-pages/man2/epoll_wait.2.html) mentioned, the return value would be zero if no file descriptor became ready during the requested timeout milliseconds.
- Add timeout to futex.
- Disable destroy operation in madvise(MADV_DONTNEED) temporarily.
- Dummy implementation of madvise(MADV_FREE).
- Ignore fcntl(F_SETLK)

In summary, the gaps are:

- [x] Extend user heap limit.
- [x] Ignore timeout in `epoll_wait` syscall.
- [ ] Implement timeout in futex wait.
- [ ] Refactor madvise(MADV_DONTNEED).
- [ ] Implement madvise(MADV_FREE).
- [ ] Implement fcntl(F_SETLK).
- [ ] Implement SigActionFlags(SA_RESETHAND).
- [ ] (Optional) Implement `/proc/self/stat`

The following are screenshots of the benchmark results(The results may be inaccurate):

Redis:

![image](https://github.com/asterinas/asterinas/assets/60682162/ba94d251-a4a1-493c-a53a-c7c306fa89e3)

sqlite3:

![image](https://github.com/asterinas/asterinas/assets/60682162/202e9ad2-6649-4361-9a18-cffd35901fd0)

### Additional Information 

Note: The SQLite3 benchmark uses the file under the `/tmp` directory in ramfs.
